src/compiler/moduleSpecifiers.ts

```diff
@@ -689,28 +700,28 @@ function tryGetModuleNameFromAmbientModule(moduleSymbol: Symbol, checker: TypeCh
      * }
      */
     // `import {c} from "m";` is valid, in which case, `moduleSymbol` is "ns", but the module name should be "m"
-    const ambientModuleDeclareCandidates = mapDefined(moduleSymbol.declarations,
-        d => {
-            if (!isModuleDeclaration(d)) return;
-            const topNamespace = getTopNamespace(d);
-            if (!(topNamespace?.parent?.parent
-                && isModuleBlock(topNamespace.parent) && isAmbientModule(topNamespace.parent.parent) && isSourceFile(topNamespace.parent.parent.parent))) return;
-            const exportAssignment = ((topNamespace.parent.parent.symbol.exports?.get("export=" as __String)?.valueDeclaration as ExportAssignment)?.expression as PropertyAccessExpression | Identifier);
-            if (!exportAssignment) return;
-            const exportSymbol = checker.getSymbolAtLocation(exportAssignment);
-            if (!exportSymbol) return;
-            const originalExportSymbol = exportSymbol?.flags & SymbolFlags.Alias ? checker.getAliasedSymbol(exportSymbol) : exportSymbol;
-            if (originalExportSymbol === d.symbol) return topNamespace.parent.parent;
-
-            function getTopNamespace(namespaceDeclaration: ModuleDeclaration) {
-                while (namespaceDeclaration.flags & NodeFlags.NestedNamespace) {
-                    namespaceDeclaration = namespaceDeclaration.parent as ModuleDeclaration;
-                }
-                return namespaceDeclaration;
+    const ambientModuleDeclareCandidates = mapDefined(moduleSymbol.declarations, d => {
+        if (!isModuleDeclaration(d)) return;
+        const topNamespace = getTopNamespace(d);
+        if (
+            !(topNamespace?.parent?.parent
+                && isModuleBlock(topNamespace.parent) && isAmbientModule(topNamespace.parent.parent) && isSourceFile(topNamespace.parent.parent.parent))
+        ) return;

```

Comment:
this seems like a weird way to avoid to the wrap-in-braces rule, but I see that it's based on the original code, and is 100x more readable because the `return` is not at the end of a long line